usaspending_api/database_scripts/etl/location_delta_view.sql

```diff
@@ -1,460 +1,241 @@
-    FROM
-        transaction_locations_cte
-    UNION
-    SELECT
-        pop_zip_string AS location,
-        pop_zip_json AS location_json
-    FROM
-        transaction_locations_cte
-    UNION
-    SELECT
-        pop_current_cd_string AS location,
-        pop_current_cd_json AS location_json
-    FROM
-        transaction_locations_cte
-    UNION
-    SELECT
-        pop_original_cd_string AS location,
-        pop_original_cd_json AS location_json
-    FROM
-        transaction_locations_cte
-    UNION
-    SELECT
-        recipient_location_country_string AS location,
-        recipient_location_country_json AS location_json
-    FROM
-        transaction_locations_cte
-    UNION
-    SELECT
-        recipient_location_state_string AS location,
-        recipient_location_state_json AS location_json
-    FROM
-        transaction_locations_cte
-    UNION
-    SELECT
-        recipient_location_city_string AS location,
-        recipient_location_city_json AS location_json
-    FROM
-        transaction_locations_cte
-    UNION
-    SELECT
-        recipient_location_county_string AS location,
-        recipient_location_county_json AS location_json
-    FROM
-        transaction_locations_cte
-    UNION
-    SELECT
-        recipient_location_zip_string AS location,
-        recipient_location_zip_json AS location_json
-    FROM
-        transaction_locations_cte
-    UNION
-    SELECT
-        recipient_location_current_cd_string AS location,
-        recipient_location_current_cd_json AS location_json
-    FROM
-        transaction_locations_cte
-    UNION
-    SELECT
-        recipient_location_original_cd_string AS location,
-        recipient_location_original_cd_json AS location_json
-    FROM
-        transaction_locations_cte
-)
+        FROM
+            rpt.transaction_search
+        RIGHT JOIN
+            state_data AS sd ON sd.code = recipient_location_state_code
+        WHERE
+            recipient_location_state_code IS NOT NULL
+            AND
+            recipient_location_congressional_code ~ '^[0-9]{2}$'
+    )
 SELECT
     ROW_NUMBER() OVER (ORDER BY location, location_json) AS id,
     location,
     location_json
 FROM
-    normalized_column_names_cte
-WHERE
-    -- Only include locations that have at least two characters
-    location ~ '[A-Z0-9].*[A-Z0-9]'
-    and
-    location_json IS NOT NULL
+    (
+        SELECT * FROM country_cte
+        UNION
+        SELECT * FROM state_cte
+        UNION
+        SELECT * FROM city_domestic
+        UNION
+        SELECT * FROM city_foreign_pop_cte
+        UNION
+        SELECT * FROM city_foreign_rl_cte
+        UNION
+        SELECT * FROM county_cte
+        UNION
+        SELECT * FROM zip_cte
+        UNION
+        SELECT * FROM current_cd_pop_cte
+        UNION
+        SELECT * FROM current_cd_rl_cte
+        UNION
+        SELECT * FROM original_cd_pop_cte
+        UNION
+        SELECT * FROM original_cd_rl_cte
+    ) AS union_all
+GROUP BY
```

Comment:
Why do you need to GROUP BY here?

Reply:
The UNION should be removing all duplicates

Reply:
I completely forgot that `UNION` will remove duplicates. I'll update this to remove the `GROUP BY`.